packages/compiler-cli/src/ngtsc/annotations/directive/src/shared.ts

```diff
@@ -187,6 +188,10 @@ export function extractDirectiveMetadata(
   const hostDirectives =
       rawHostDirectives === null ? null : extractHostDirectives(rawHostDirectives, evaluator);
 
+  if (hostDirectives !== null) {
+    referencesRegistry.add(clazz, ...hostDirectives.map(hostDir => hostDir.directive));
```

Comment:
Maybe a comment here:
```suggestion
    // The template type-checker will need to import host directive types, so add them
    // as referenced by `clazz`. This will ensure that libraries are required to export
    // host directives which are visible from publicly exported components.
    referencesRegistry.add(clazz, ...hostDirectives.map(hostDir => hostDir.directive));
```